lib/pages/settings_dashboard/settings/settings.dart

```diff
@@ -86,11 +86,40 @@ class SettingsController extends State<Settings> with ConnectPageMixin {
         OkCancelResult.cancel) {
       return;
     }
-    final matrix = Matrix.of(context);
     if (PlatformInfos.isMobile) {
+      await _logoutActionsOnMobile();
+    } else {
+      await _logoutActionsOnWeb();
+    }
+  }
+
+  Future<void> _logoutActionsOnMobile() async {
+    try {
       await tryLogoutSso(context);
+    } catch (e) {
+      Logs().e('SettingsController()::_logoutActionsOnMobile - error: $e');
+      return;
+    }
+    if (matrix.canDeleteToMDatabase == true) {
```

Comment:
you can replace it with `if (matrix.canDeleteToMDatabase) {` since `matrix` is not nullable